app/javascript/mastodon/features/compose/components/compose_form.js

```diff
@@ -280,9 +284,8 @@ class ComposeForm extends ImmutablePureComponent {
         <div className='compose-form__publish'>
           <div className='compose-form__publish-button-wrapper'>
             <Button
-              type="submit"
+              type='submit'
               text={publishText}
-              onClick={this.handleSubmit}
```

Comment:
Since we are removing onClick here, this raises:

```
Uncaught TypeError: this.props.onClick is not a function
    at Button.handleClick
```

We should make the `onClick` optional inside `Button`. E.g.:


```js
  handleClick = (e) => {
    if (!this.props.disabled && this.props.onClick) {
      this.props.onClick(e);
    }
  }
```